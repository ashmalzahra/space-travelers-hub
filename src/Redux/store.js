import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './Rockets/RocketsSlice';
import missionsSlice from './MissionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice.reducer,
  },
});

export default store;
