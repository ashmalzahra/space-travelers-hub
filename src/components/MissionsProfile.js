<<<<<<< HEAD
import React from 'react';
=======
>>>>>>> 9add45350724ad927b921ddfda852e38054be055
import { useSelector } from 'react-redux';

const MissionsProfile = () => {
  const missions = useSelector((state) => state.missions.missions);
  const reservedMissions = missions.filter((x) => x.reserved);

  return (
    <div className="reserve-container">
      <h1 className="profile-missions-title">My Missions</h1>
      <div className="profile-missions">
        {reservedMissions.length === 0 && <p className="no-reserve">No Mission Reserved</p>}
        {reservedMissions.map((x) => (
          <div key={x.mission_id} className="reserve-item">{x.mission_name}</div>
        ))}

      </div>
    </div>
  );
};

export default MissionsProfile;
