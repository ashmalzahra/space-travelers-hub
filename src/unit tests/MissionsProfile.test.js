import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import MissionsProfile from '../components/MissionsProfile';
import store from '../Redux/store';

it('render', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MissionsProfile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders content', () => {
  const component = render(
    <Provider store={store}>
      <MissionsProfile />
    </Provider>,
  );
  expect(component.container).toHaveTextContent('My Missions');
});
