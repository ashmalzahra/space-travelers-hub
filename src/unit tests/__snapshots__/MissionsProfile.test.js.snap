// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="reserve-container"
>
  <h1
    className="profile-missions-title"
  >
    My Missions
  </h1>
  <div
    className="profile-missions"
  >
    <p
      className="no-reserve"
    >
      No Mission Reserved
    </p>
  </div>
</div>
`;
